src/main/java/service/FruitShopService.java

```diff
@@ -0,0 +1,36 @@
+package service;
+
+import exception.OperationException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import model.OperationType;
+import strategy.OperationHandler;
+
+public class FruitShopService {
+    private final Map<String, Integer> fruitInventory = new HashMap<>();
```

Comment:
Consider using a more specific data structure that suits the needs of fruit inventory management, such as LinkedHashMap to preserve insertion order or TreeMap for sorted order if necessary.